src/components/BottomBar/SkyBrowser/SkyBrowserTabs.jsx

```diff
@@ -14,74 +14,165 @@ import {
   reloadPropertyTree,
 } from '../../../api/Actions';
 
-class SkyBrowserTabs extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isShowingInfoButtons: [false, false, false, false, false],
-      showSettings: false,
-      messageCounter: 0
-    };
-    this.createTabs = this.createTabs.bind(this);
-    this.createImageList = this.createImageList.bind(this);
-    this.showTooltip = this.showTooltip.bind(this);
-    this.hideTooltip = this.hideTooltip.bind(this);
-    this.createButtons = this.createButtons.bind(this);
-    this.toggleShowSettings = this.toggleShowSettings.bind(this);
-    this.setImageLayerOrder = this.setImageLayerOrder.bind(this);
-    this.createTargetBrowserPair = this.createTargetBrowserPair.bind(this);
-    this.removeTargetBrowserPair = this.removeTargetBrowserPair.bind(this);
+function SkyBrowserTabs({ 
+  setCurrentTabHeight,
+  passMessageToWwt,
+  setWwtRatio,
+  activeImage,
+  currentBrowserColor,
+  selectImage,
+  maxHeight,
+  minHeight,
+  height,
+  setBorderRadius,
+  imageCollectionIsLoaded
+}) {
+  const [isShowingInfoButtons, setIsShowingInfoButtons] =
+    React.useState([false, false, false, false, false]);
+  const [showSettings, setShowSettings] = React.useState(false);
+  const [messageCounter, setMessageCounter] = React.useState(0);
+
+  const infoButton = React.useRef(null);
+  const tabsDiv = React.useRef(null);
+
+  const browsers = useSelector((state) => {
+    return state.skybrowser.browsers
+  }, shallowEqual);
+
+  const luaApi = useSelector((state) => {
+    return state.luaApi
+  }, shallowEqual);
+
+  const selectedBrowserId = useSelector((state) => {
+    return state.skybrowser.selectedBrowserId
+  }, shallowEqual);
+
+  const data = useSelector((state) => {
+    const browser = browsers[selectedBrowserId];
+    if (!state.skybrowser.imageList || !browser) {
+      return [];
+    }
+    const images = browser.selectedImages;
+    if (!images) {
+      return [];
+    }
+    const indices = Object.values(images);
+    return indices.map(index => state.skybrowser.imageList[index.toString()]);
+  }, shallowEqual);
+
+  const dispatch = useDispatch();
```

Comment:
This is how we will want to work with Redux and dispatches for all the functional components in the future, right?

How about putting it at the top of the function or somehting? Not sure what a good code structure for dealing with this is. But previously I really liked that all the dispatch related stuff was always at the end of the file. This line seems a bit randomly placed